### PR TITLE
feat(border-radius): missing test cases from another PR

### PR DESCRIPTION
**Pull request by renoirb**
Making compromises between Draft PR #12801 from which I was throwing ideas, and current PR #12886

## Description

* Instead of writing possible values of a type, write once, and create a type based on it
* Missing test cases

## How Has This Been Tested?

Yes, and is adding more.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [ ] The PR title is no longer than 64 characters.
- [ ] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/composables/border-radius.ts [hunk 1]
```diff
@@ -37,7 +39,7 @@ export const makeBorderRadiusProps = propsFactory({
 export function useBorderRadius (props: BorderRadiusProps) {
   const borderRadiusClasses = computed(() => {
     const classes = []
-    const rounded = props.rounded
+    const rounded = props && 'rounded' in props ? props.rounded : null
```
Comment (renoirb) [inline packages/vuetify/src/composables/border-radius.ts:42]: There was this missing possibility

File: packages/vuetify/src/composables/__tests__/border-radius.spec.ts [hunk 2]
```diff
@@ -21,10 +21,27 @@ describe('border-radius.ts', () => {
...
+    // Rounded only
+    [{ rounded: true }, ['rounded']],
+    [{ rounded: '' }, ['rounded']],
+    // When should return nothing
+    [{ rounded: 1 }, []],
+    [{ rounded: { toString: () => 'pill' } }, []],
+    // All strings are accepted?
+    [{ rounded: 'foo-bar-bazz-buzz' }, ['rounded-foo-bar-bazz-buzz']],
+    [{ rounded: '!' }, ['rounded-!']],
+    [{ rounded: 'sm-3 4' }, ['rounded-sm-3', 'rounded-4']],
```
Comment (renoirb) [inline packages/vuetify/src/composables/__tests__/border-radius.spec.ts:44]: Moar test cases to ensure we cover possible values.

File: packages/vuetify/src/composables/__tests__/border-radius.spec.ts [hunk 3]
```diff
@@ -20,11 +20,28 @@ describe('border-radius.ts', () => {
...
     [{ rounded: false }, ['rounded-0']],
+    [{ rounded: 0 }, ['rounded-0']],
+    // Rounded with a word
+    [{ rounded: 'circle' }, ['rounded-circle']],
+    [{ rounded: 'shaped' }, ['rounded-shaped']],
+    [{ rounded: 'pill' }, ['rounded-pill']],
+    // All strings are accepted?
+    [{ rounded: 'foo-bar-bazz-buzz' }, ['rounded-foo-bar-bazz-buzz']],
+    [{ rounded: '!' }, ['rounded-!']],
+    [{ rounded: 'sm-3 4' }, ['rounded-sm-3', 'rounded-4']],
```
Comment (renoirb) [inline packages/vuetify/src/composables/__tests__/border-radius.spec.ts:44]: Moar test cases to ensure we cover possible values.

File: packages/vuetify/src/composables/border-radius.ts (hunk 1 above)
Comment (KaelWD) [reply packages/vuetify/src/composables/border-radius.ts:42]: What was? If `'rounded' in props` is false then `props.rounded` is undefined and the `rounded == null` condition matches.

File: packages/vuetify/src/composables/__tests__/border-radius.spec.ts [hunk 4]
```diff
@@ -20,11 +20,28 @@ describe('border-radius.ts', () => {
   })
 
   it.each([
+    // When should return nothing
     [{}, []],
-    [{ rounded: 'tile' }, ['rounded-0']],
+    [undefined, []],
```
Comment (KaelWD) [inline packages/vuetify/src/composables/__tests__/border-radius.spec.ts:25]: `props` is always at least `{}`

File: packages/vuetify/src/composables/border-radius.ts [hunk 5]
```diff
@@ -24,7 +26,7 @@ const radius: BorderRadiusProp[] = [
   'pill',
   'shaped',
   'tile',
-]
+] as const
```
Comment (KaelWD) [inline packages/vuetify/src/composables/border-radius.ts:29]: :+1:

File: packages/vuetify/src/composables/border-radius.ts (hunk 1 above)
Comment (renoirb) [reply packages/vuetify/src/composables/border-radius.ts]: Fair enough, adjusted

**Review by johnleider — APPROVED** [review]

**Pull request merged**